A4A shadow styling fix.

tdrl opened the pull request:
A4A was not setting AMP styling within the shadow DOM for legitimate AMP ads rendered there.  As a consequence, the elements were inferring 0 sizing and never rendering.  This adds styling by copying from the document head.  This is a short-term fix, as we work out how to handle it more cleanly.

cramforce commented on extensions/amp-a4a/0.1/amp-a4a.js:
```diff
@@ -545,6 +545,9 @@ export class AmpA4A extends AMP.BaseElement {
             // Finally, add body and re-formatted CSS styling to the shadow root.
             const shadowRoot =
                 this.element.shadowRoot || this.element.createShadowRoot();
+	    const style = this.getWin().document.querySelector(
```
Nit: indent is off

tdrl replied:
Oops.  Fixed.

jridgewell commented:
LGTM. Is this the right approach?

cramforce commented:
@jridgewell We discussed this in a separate thread.

Basically no, but the right thing for today :)

The right long term fix is to lift code from Dima's work in #3742

cramforce commented:
@tdrl Unfortunately this fails your tests. Please update them.

Thanks!

tdrl commented:
@cramforce D'oh!  It's what I get for going too fast.  :-P

Fixed: Added some defensiveness to A4A and updated tests to insert a synthetic `<style amp-runtime>` tag.

jridgewell commented:
LGTM. Is this ready to ship?

cramforce commented:
🚢

cramforce commented:
@jasti Should we merge this up into the canary that we just pushed? Or is next week OK?
